preconditions/src/main/java/com/palantir/logsafe/exceptions/SafeIllegalArgumentException.java

```diff
@@ -26,6 +26,7 @@
     private final List<Arg<?>> arguments;
 
     public SafeIllegalArgumentException() {
+        super("");
```

Comment:
Tiny bit surprised this is necessary.... is this just to allow the `hasMessage("")` thing?

Reply:
Yeh, so I realised the message is otherwise null, and then the `getLogMessage()` returns null, which I would assume is not OK (the interface does not explicitly specify it as `Nullable`).  I could instead change the `getLogMessage()` function to `return msg == null ? "" : msg`, but then I have to override all of the assert functions which are message dependent to use the log message rather than the actual message, which feels wrong.

Any suggestions or thoughts welcome.

Reply:
:+1: this seems sensible